overlord/configstate/configcore/vitality.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/snap/naming"
+)
+
+const vitalityOpt = "resilience.vitality-hint"
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core."+vitalityOpt] = true
+}
+
+func handleVitalityConfiguration(tr config.Conf) error {
+	// XXX: rewrite/restart changed configs
+
+	return nil
+}
+
+func validateVitalitySettings(tr config.Conf) error {
+	option, err := coreCfg(tr, vitalityOpt)
+	if err != nil {
+		return err
+	}
+	if option == "" {
+		return nil
+	}
+
+	for _, snapName := range strings.Split(option, ",") {
+		if err := naming.ValidateInstance(snapName); err != nil {
+			return fmt.Errorf("cannot set %q: %v", vitalityOpt, err)
+		}
+	}
+
```

Comment:
should we also validate that there's not more than 1900 snap names in this key since we set `OOMScoreAdjust = -(899 - rank)`, and if rank is 1900 this becomes +1001. It appears that the kernel will just not let systemd specify that high of a score, but what's less clear to me is what systemd does if that fails?

Reply:
We actually should be a bit more strict, I would say less not support more than 100 snaps in the key

Reply:
Added this